pool_automation/roles/aws_manage/library/test_stateful_set.py

```diff
@@ -93,21 +103,60 @@ def test_find_ubuntu_image(ec2):
     assert 'UNSUPPORTED' not in image.description
 
 
-def test_create_instances(ec2):
+def test_AwsEC2Launcher(ec2):
+    launcher = AwsEC2Launcher()
     params = PARAMS._replace(role='test_create')
-    instances = create_instances(ec2, params, 2)
+    instances = launcher.launch(params, 2, ec2=ec2)
 
     assert len(instances) == 2
+    assert len(launcher._ids) > 0
+
     for instance in instances:
+        instance.reload()
+        assert instance.state['Name'] == 'pending'
+
+    launcher.wait()
+    assert len(launcher._ids) == 0
+
+    for instance in instances:
+        instance.reload()
```

Comment:
We already store instances in launcher, can we call `reload` inside `wait` to make API more convenient?

Reply:
Waiter uses boto3 client api, reload is from resource api. And in any case we shouldn't rely on implementation details: here we should be sure that instance state is synced before any checks. Thus I don't think it's a good idea.